internal/pkg/service/common/configmap/visit.go

```diff
@@ -38,6 +41,7 @@ type VisitContext struct {
 	Validate string
 }
 
+// Visit each nested structure field.
 func Visit(value reflect.Value, cfg VisitConfig) error {
 	vc := &VisitContext{}
```

Comment:
I refactored the `configmap.Visit` function that already visits configuration structure.

But I need to visit also `nil` pointers, in the `ConfigPatch` structures, to get the full list of all keys.
See test in the next commit.

-----------

Original tests are unmodified. 

In the tests, all logic is covered, missing coverage is for different base types: `int8`, `int16`, ... and some rare errors.

![image](https://github.com/keboola/keboola-as-code/assets/19371734/acbc433b-633f-43c2-b779-30ed8e586197)

-----------------

The code contains same blocks, but they were reorganized, and handling for `nil` and `invalid` cases have been added.